### DISCUSSION TITLE
How to deal with initializing

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#10375] [General]
I've noticed that sometimes a lambda inside a react statement will no longer be called if it throws an exception during initialization.  The system is not displaying exceptions when thrown from a react lambda, but I can see it if I explicitly do a `try...catch` and put a breakpoint or just run the debugger.

This results because have a number of properties which are connected to each other in xaml.  The basic `AvaloniaProperty` generally sets the backing variable to something neutral like an `INVALID` enum, or `string.Empty`, or `0.0f`, or `0L`, etc.  Something that is as neutral and meaningless as possible.

I then try to set the actual, used value, either from the xaml, like `<MyControl SomeProperty="15"/>`, or in the `OnApplyTemplate` method.

If this property is connected to a few other objects in xaml, and that logic generates values that are used by other objects, etc., then there is a cascade of events that happen at startup, including the initial updates from all the `this.WhenAnyValue(...).Subscribe(...)`.

I can make it so the react lambda is properly called by dealing with `INVALID` values gently, but this is only needed for initialization.  I'd like for `INVALID` generally to cause a proper exception during normal runtime.

So the question is, as this style of programming goes, Is it common to have to put all sorts of gentle forgiveness in my methods when they receive an uninitialized value such as my `INVALID` enum, or let's say a nullable that is set to null at startup, but later gets a value?  Normally this would be cause to throw an exception, but I find that I only get a proper value after things have settled and the various controls are up and running.

So what is the correct pattern here?  Looking for high level guidance. 

Thank you

**Comment by timunie** [comment]
From the snippets above hard to see  what's wrong. Would be good to see the code needed to reproduce your issue. Can you share a minimal sample with us?
